libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -305,8 +306,9 @@ struct eosvmoc_tier {
       platform_timer& main_thread_timer;
       const wasm_interface::vm_type wasm_runtime_time;
       const wasm_interface::vm_oc_enable eosvmoc_tierup;
-      std::atomic<uint64_t> executing_action_id{1}; // monotonic increasing for each action apply
+      large_atomic<digest_type> executing_code_hash{};
```

Comment:
what about `sha256` prevents it being used as a `atomic<T>`?

Reply:
`undefined reference to `__atomic_store'`

Reply:
That just means we're not linking to `libatomic` (which is kind of surprising given the maturity of spring that it was never needed previously), I thought maybe something else about sha256 was non trivial and preventing usage of it with std::atomic.

Reply:
Not sure if there is anything else that prevents it. Should we start linking with `libatomic`?